### PR TITLE
Add provider discriminator to common/reducer.js and actions

### LINKED ISSUE #2829: Add a local/unsplash discriminator attribute to context actions
## Task Description

Add a local/unsplash discriminator attribute to context actions.
The discriminator will be used to route common media actions to the right provider reducer (local, unsplash, coverr etc).

### PR DESCRIPTION
**Pull request by cvolzke4**
## Summary

Add provider discriminator to common/reducer.js and actions.
This allows the common actions and reducer to be shared between local and media3p for different providers.

## Testing Instructions

- Open the media pane to verify that it appears.
- Scroll down to verify a page of results in returned.
- Upload an image successfully.
- Upload a video successfully.

### CODE REVIEW

File: assets/src/edit-story/app/media/local/actions.js [hunk 1]
```diff
@@ -20,12 +20,17 @@
...
 import * as types from '../types';
 
-export const fetchMediaStart = common.fetchMediaStart;
-export const fetchMediaSuccess = common.fetchMediaSuccess;
-export const fetchMediaError = common.fetchMediaError;
-export const setNextPage = common.setNextPage;
-export const updateMediaElement = common.updateMediaElement;
-export const deleteMediaElement = common.deleteMediaElement;
+export const fetchMediaStart = (dispatch) =>
+  common.fetchMediaStart(dispatch, 'local');
```
Comment (ezxyz) [inline assets/src/edit-story/app/media/local/actions.js:24]: Extract 'local' to a constant.

File: assets/src/edit-story/app/media/local/test/reducer.js [hunk 2]
```diff
@@ -25,8 +25,61 @@ import { renderHook } from '@testing-library/react-hooks';
 import useMediaReducer from '../../useMediaReducer';
 import reducer from '../reducer';
 import * as actionsToWrap from '../actions';
```
Comment (ezxyz) [inline assets/src/edit-story/app/media/local/test/reducer.js]: could you specify these are localActionsToWrap ?  I found that its a bit confusing that we have two different actions files we're importing from.

File: assets/src/edit-story/app/media/local/test/reducer.js [hunk 3]
```diff
@@ -25,8 +25,61 @@ import { renderHook } from '@testing-library/react-hooks';
...
+      })
+    );
+  });
+
+  it('should assign isMediaLoading=true on fetchMediaStart', () => {
+    const { result } = renderHook(() =>
+      useMediaReducer(reducer, actionsToWrap)
+    );
+
+    result.current.actions.fetchMediaStart({ pageToken: 'page2' });
```
Comment (ezxyz) [inline assets/src/edit-story/app/media/local/test/reducer.js:54]: can this be renamed localFetchMediaStart so its clearer its local?

**Review by ezxyz — APPROVED** [review]

File: assets/src/edit-story/app/media/local/actions.js (hunk 1 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/local/actions.js:24]: Though constants.local in place of 'local' isn't an improvement to type safety, I'd like to add a constant as a part of issue https://github.com/google/web-stories-wp/issues/2804.

File: assets/src/edit-story/app/media/local/test/reducer.js (hunk 2 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/local/test/reducer.js]: Done.

File: assets/src/edit-story/app/media/local/test/reducer.js (hunk 3 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/local/test/reducer.js:54]: Does this mean other local actions that don't appear in media3p should be named this way? eg:

- setProcessing -> localSetProcessing
- setProcessed -> localSetProcessed
- setMediaType -> localSetMediaType

The scope of the function should be made clear by the folder it's in, media/local or media/media3p.

**Pull request merged**